app/Models/Image.php

```diff
@@ -154,6 +125,37 @@ public static function upload(UploadedFile $file, $user, bool $forceReupload = f
         }
     }
 
+    /**
+     * Converts the image into the specified cover type to the specified path.
+     *
+     * @param File $image The image file to be processed
+     * @param string $path The path to save the processed image file
+     * @param int $coverType The type to process the image to
+     */
+    public static function processFile(File $image, string $path, $coverType) {
+        if ($coverType['id'] === self::ORIGINAL && $image->getMimeType() === 'image/jpeg') {
+            $command = 'cp "'.$image->getPathname().'" '.$path;
+        } else {
+            // ImageMagick options reference: http://www.imagemagick.org/script/command-line-options.php
+            $command = 'convert 2>&1 "'.$image->getPathname().'" -background white -alpha remove -alpha off -strip';
+
+            if ($image->getMimeType() === 'image/jpeg') {
+                $command .= ' -quality 100 -format jpeg';
+            } else {
+                $command .= ' -quality 95 -format png';
+            }
+
+            if (isset($coverType['geometry'])) {
+                $command .= " -gravity center -thumbnail ${coverType['geometry']} -extent ${coverType['geometry']}";
+            }
+
+            $command .= ' "'.$path.'"';
+        }
+
+        External::execute($command);
```

Comment:
What happens if/when this call fails? Does the command's stderr get logged anywhere?

Reply:
The external class is something you wrote, that appears to log an error to stderr if the command fails. To be honest, though, this should probably be rewritten some other time so that it uses the PHP ImageMagick extension.

Reply:
I looked over at that code and I don't like it. 😢 It should throw up an exception if it fails but that'll be a task in its own right. I'll leave it be where this particular merge is concerned.